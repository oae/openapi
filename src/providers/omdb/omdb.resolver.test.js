const resolvers = require('./omdb.resolver.js');
const contextCreator = require('./context');

const info = `{
    title
    year
    imdb {
      id
      rating
      votes
    }
    metacritic {
      rating
    }
    rottenTomatoes {
      rating
    }
    rated
    released
    runtime
    genre
    director
    writer
    actors
    plot
    language
    country
    awards
    poster
    boxOffice
    production
    website
  }`;

const createContext = (...contextArgs) => {
  let context = null;

  return {
    async getContext(name) {
      if (!context) {
        context = await contextCreator(...contextArgs);
      }

      return context;
    },
  };
};

describe('omdb', () => {
  it('should return movies with title', async () => {
    const context = createContext();

    const result = await resolvers.Query.movies(
      {},
      { title: 'Inception', limit: 5 },
      context,
      info
    );
    expect(result).toMatchSnapshot();
  });

  it('should return single movie', async () => {
    const context = createContext();

    const result = await resolvers.Query.movie({}, { imdbId: 'tt1375666' }, context, info);
    expect(result).toMatchSnapshot();
  });
});
