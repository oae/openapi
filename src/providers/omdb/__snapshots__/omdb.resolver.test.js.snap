// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omdb should return movies with title 1`] = `
Array [
  Object {
    "Actors": "Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page, Tom Hardy",
    "Awards": "Won 4 Oscars. Another 152 wins & 204 nominations.",
    "BoxOffice": "$292,568,851",
    "Country": "USA, UK",
    "DVD": "07 Dec 2010",
    "Director": "Christopher Nolan",
    "Genre": "Action, Adventure, Sci-Fi, Thriller",
    "Language": "English, Japanese, French",
    "Metascore": "74",
    "Plot": "A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a CEO.",
    "Poster": "https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg",
    "Production": "Warner Bros. Pictures",
    "Rated": "PG-13",
    "Ratings": Array [
      Object {
        "Source": "Internet Movie Database",
        "Value": "8.8/10",
      },
      Object {
        "Source": "Rotten Tomatoes",
        "Value": "86%",
      },
      Object {
        "Source": "Metacritic",
        "Value": "74/100",
      },
    ],
    "Released": "16 Jul 2010",
    "Response": "True",
    "Runtime": "148 min",
    "Title": "Inception",
    "Type": "movie",
    "Website": "http://inceptionmovie.warnerbros.com/",
    "Writer": "Christopher Nolan",
    "Year": "2010",
    "imdbID": "tt1375666",
    "imdbRating": "8.8",
    "imdbVotes": "1,755,015",
  },
  Object {
    "Actors": "Leonardo DiCaprio, Joseph Gordon-Levitt, Lukas Haas",
    "Awards": "N/A",
    "BoxOffice": "N/A",
    "Country": "USA",
    "DVD": "N/A",
    "Director": "Ian Kirby",
    "Genre": "Animation, Short, Sci-Fi",
    "Language": "English",
    "Metascore": "N/A",
    "Plot": "This Inception prequel unfolds courtesy of a beautiful Motion Comic, and explains how Cobb, Arthur and Nash were enlisted by Cobol Engineering. Diehard fans of the film will be especially interested in this one.",
    "Poster": "https://m.media-amazon.com/images/M/MV5BMjE0NGIwM2EtZjQxZi00ZTE5LWExN2MtNDBlMjY1ZmZkYjU3XkEyXkFqcGdeQXVyNjMwNzk3Mjk@._V1_SX300.jpg",
    "Production": "N/A",
    "Rated": "N/A",
    "Ratings": Array [
      Object {
        "Source": "Internet Movie Database",
        "Value": "7.8/10",
      },
    ],
    "Released": "07 Dec 2010",
    "Response": "True",
    "Runtime": "15 min",
    "Title": "Inception: The Cobol Job",
    "Type": "movie",
    "Website": "N/A",
    "Writer": "Christopher Nolan (based on characters by)",
    "Year": "2010",
    "imdbID": "tt5295894",
    "imdbRating": "7.8",
    "imdbVotes": "1,138",
  },
  Object {
    "Actors": "Rick Edwards",
    "Awards": "N/A",
    "BoxOffice": "N/A",
    "Country": "UK",
    "DVD": "N/A",
    "Director": "Steve Kemsley",
    "Genre": "Documentary",
    "Language": "English",
    "Metascore": "N/A",
    "Plot": "N/A",
    "Poster": "N/A",
    "Production": "N/A",
    "Rated": "N/A",
    "Ratings": Array [
      Object {
        "Source": "Internet Movie Database",
        "Value": "7.2/10",
      },
    ],
    "Released": "16 Jul 2010",
    "Response": "True",
    "Runtime": "N/A",
    "Title": "Inception: 4Movie Premiere Special",
    "Type": "movie",
    "Website": "N/A",
    "Writer": "Steven Vinacour",
    "Year": "2010",
    "imdbID": "tt1686778",
    "imdbRating": "7.2",
    "imdbVotes": "21",
  },
  Object {
    "Actors": "Bret Hart, Jeff Jarrett, Brian James, David Heath",
    "Awards": "N/A",
    "BoxOffice": "N/A",
    "Country": "Australia",
    "DVD": "N/A",
    "Director": "N/A",
    "Genre": "N/A",
    "Language": "English",
    "Metascore": "N/A",
    "Plot": "N/A",
    "Poster": "https://m.media-amazon.com/images/M/MV5BNTEyNGJjMTMtZjZhZC00ODFkLWIyYzktN2JjMTcwMmY5MDJlXkEyXkFqcGdeQXVyNDkwMzY5NjQ@._V1_SX300.jpg",
    "Production": "N/A",
    "Rated": "N/A",
    "Ratings": Array [
      Object {
        "Source": "Internet Movie Database",
        "Value": "6.0/10",
      },
    ],
    "Released": "26 Oct 2001",
    "Response": "True",
    "Runtime": "N/A",
    "Title": "WWA: The Inception",
    "Type": "movie",
    "Website": "N/A",
    "Writer": "Jeremy Borash",
    "Year": "2001",
    "imdbID": "tt0311992",
    "imdbRating": "6.0",
    "imdbVotes": "20",
  },
  Object {
    "Actors": "Erfan Refahatnia",
    "Awards": "N/A",
    "BoxOffice": "N/A",
    "Country": "Iran",
    "DVD": "N/A",
    "Director": "Danial Hajibarat",
    "Genre": "Short, Drama, Mystery",
    "Language": "Persian",
    "Metascore": "N/A",
    "Plot": "Young man wants to do something important,It has a lot of risk for him But he must make a decision,Though everyone around them opposes doing so,The boy decides that...",
    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BYWJmYWJmNWMtZTBmNy00M2MzLTg5ZWEtOGU5ZWRiYTE0ZjVmXkEyXkFqcGdeQXVyNzkyOTM2MjE@._V1_SX300.jpg",
    "Production": "N/A",
    "Rated": "N/A",
    "Ratings": Array [
      Object {
        "Source": "Internet Movie Database",
        "Value": "10.0/10",
      },
    ],
    "Released": "01 Feb 2014",
    "Response": "True",
    "Runtime": "11 min",
    "Title": "Inception",
    "Type": "movie",
    "Website": "N/A",
    "Writer": "Danial Hajibarat",
    "Year": "2014",
    "imdbID": "tt7321322",
    "imdbRating": "10.0",
    "imdbVotes": "14",
  },
]
`;

exports[`omdb should return single movie 1`] = `
Object {
  "Actors": "Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page, Tom Hardy",
  "Awards": "Won 4 Oscars. Another 152 wins & 204 nominations.",
  "BoxOffice": "$292,568,851",
  "Country": "USA, UK",
  "DVD": "07 Dec 2010",
  "Director": "Christopher Nolan",
  "Genre": "Action, Adventure, Sci-Fi, Thriller",
  "Language": "English, Japanese, French",
  "Metascore": "74",
  "Plot": "A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a CEO.",
  "Poster": "https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg",
  "Production": "Warner Bros. Pictures",
  "Rated": "PG-13",
  "Ratings": Array [
    Object {
      "Source": "Internet Movie Database",
      "Value": "8.8/10",
    },
    Object {
      "Source": "Rotten Tomatoes",
      "Value": "86%",
    },
    Object {
      "Source": "Metacritic",
      "Value": "74/100",
    },
  ],
  "Released": "16 Jul 2010",
  "Response": "True",
  "Runtime": "148 min",
  "Title": "Inception",
  "Type": "movie",
  "Website": "http://inceptionmovie.warnerbros.com/",
  "Writer": "Christopher Nolan",
  "Year": "2010",
  "imdbID": "tt1375666",
  "imdbRating": "8.8",
  "imdbVotes": "1,755,015",
}
`;
